dds_web/utils.py

```diff
@@ -749,3 +749,64 @@ def use_sto4(unit_object, project_object) -> bool:
 
     flask.current_app.logger.info(f"{project_id_logging}sto2")
     return False
+
+
+def add_uploaded_files_to_db(proj_in_db, log):
+    from dds_web import db
+    from dds_web.api.api_s3_connector import ApiS3Connector
+
+    errors = {}
+    files_added = []
+    for file, vals in log.items():
+        status = vals.get("status")
+        if not status or not status.get("failed_op") == "add_file_db":
+            errors[file] = {"error": "Incorrect 'failed_op'."}
+            continue
+
+        with ApiS3Connector(project=proj_in_db) as s3conn:
+            try:
+                _ = s3conn.resource.meta.client.head_object(
+                    Bucket=s3conn.project.bucket, Key=vals["path_remote"]
+                )
+            except botocore.client.ClientError as err:
+                if err.response["Error"]["Code"] == "404":
+                    errors[file] = {"error": "File not found in S3", "traceback": err.__traceback__}
+            else:
+                try:
+                    file_object = models.File.query.filter(
+                        sqlalchemy.and_(
+                            models.File.name == sqlalchemy.func.binary(file),
+                            models.File.project_id == proj_in_db.id,
+                        )
+                    ).first()
+                    if file_object:
+                        errors[file] = {"error": "File already in database."}
```

Comment:
This is to fix the `--overwrite` issue
```suggestion
                    if file_object:
                        if vals.get("overwrite", False):
                            # TODO: Update the info here instead of creating a new file object
                        else:
                            errors[file] = {"error": "File already in database."}
```

Reply:
Unfortunately not, we probably need to duplicate some code.. We don't want to change anything in `NewFile.put` atm, but we could copy the code from it and put it into a utils function. 

For example:
**In utils.py**
```python
def new_file_version(file, new_info):
    """Create new version of a file."""
    from dds_web import db
    import dds_web.utils

    # Get project 
    project = file.project

    # Get version row
    current_file_version = models.Version.query.filter(
        sqlalchemy.and_(
            models.Version.active_file == sqlalchemy.func.binary(file.id),
            models.Version.time_deleted.is_(None),
        )
    ).all()
    if len(current_file_version) > 1:
        flask.current_app.logger.warning(
            "There is more than one version of the file "
            "which does not yet have a deletion timestamp."
        )

    # Same timestamp for deleted and created new file
    new_timestamp = dds_web.utils.current_time()

    # Overwritten == deleted/deactivated
    for version in current_file_version:
        if version.time_deleted is None:
            version.time_deleted = new_timestamp

    # Update file info
    file.subpath = new_info.get("subpath")
    file.size_original = new_info.get("size")
    file.size_stored = new_info.get("size_processed")
    file.compressed = new_info.get("compressed")
    file.salt = new_info.get("salt")
    file.public_key = new_info.get("public_key")
    file.time_uploaded = new_timestamp
    file.checksum = new_info.get("checksum")

    # New version
    new_version = models.Version(
        size_stored=new_info.get("size_processed"),
        time_uploaded=new_timestamp,
        active_file=file.id,
        project_id=project,
    )

    # Update foreign keys and relationships
    project.file_versions.append(new_version)
    file.versions.append(new_version)

    db.session.add(new_version)
    db.session.commit()
```

**In `add_uploaded_files_to_db`**
```python
                    if file_object:
                        if vals.get("overwrite", False):
                            new_file_version(file = file_object, new_info=vals)
                        else:    
                            errors[file] = {"error": "File already in database."}
```